packages/actions/src/Concerns/CanOpenModal.php

```diff
@@ -434,7 +434,8 @@ public function prepareModalAction(StaticAction $action): StaticAction
 
         if (
             ($this instanceof HasRecord) &&
-            ($action instanceof HasRecord)
+            ($action instanceof HasRecord) &&
+            invade($action)->record === null
```

Comment:
Please introduce a new method called `hasRecord()` in InteractsWithRecord and HasRecord, which returns true if the record is not null. So that we do not need to invade the object